src/Router/ParameterResolvingRouter.php

```diff
@@ -63,6 +64,27 @@ public function getRouteCollection()
      */
     public function generate($name, $parameters = [], $referenceType = UrlGeneratorInterface::ABSOLUTE_PATH)
     {
+        $resolvedParameters = $this->resolverCollection->resolve($parameters);
+        $unresolved = [];
+
+        foreach ($resolvedParameters as $key => $parameter) {
+            // sanity check for parameters that could not be resolved at all
+            if (is_object($parameter)) {
+                $unresolved[$key] = get_class($parameter);
+                continue;
+            }
+
+            // sanity check for parameters that could not be resolved into a usable value
+            if ('' === (string) $parameter) {
```

Comment:
Arrays aren't allowed in the first place, they would've caused an issue in the generate method already. I could add an array check as well. I think that should cover most cases :thinking: